dbt_subprojects/daily_spellbook/models/_sector/dex/pools/optimism/dex/dex_pools_optimism_balances.sql

```diff
@@ -0,0 +1,181 @@
+{{
+  config(
+    schema = 'pools_optimism',
+    alias = 'balances',
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['pool_address', 'snapshot_day'],
+    incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.snapshot_day')]
+  )
+}}
+
+WITH uniswap_v3_op_addresses AS (
+  SELECT
+    pool AS address,
+    token0,
+    token1,
+    0x4200000000000000000000000000000000000042 AS token_address,
+    creation_block_time AS creation_time,
+    'uniswap' AS protocol_name,
+    'v3' AS version
+  FROM 
+    {{ source('uniswap_v3_optimism', 'pools') }}
+  WHERE
+    token0 = 0x4200000000000000000000000000000000000042
+    OR token1 = 0x4200000000000000000000000000000000000042
```

Comment:
```suggestion
  WHERE
    (token0 = 0x4200000000000000000000000000000000000042
    OR token1 = 0x4200000000000000000000000000000000000042)
    {% if is_incremental() %}
    AND {{ incremental_predicate('creation_block_time') }}
    {% endif %}
```

Reply:
please add this change to all source CTE reads, only commenting here to keep clean